dfttk/eos_fit.py

```diff
@@ -1264,88 +1260,18 @@ def fit_to_all_eos(df: pd.DataFrame) -> tuple[pd.DataFrame, pd.DataFrame]:
-) -> pd.DataFrame:
-    """
-    The input_files should be a list of strings containing the file names of the input files
-
-    the name of each input files should end in'_x' where x is the config_name ex:
-        str_0, str_1, str_2, ...
-        volume_energy_0, volume_energy_1, volume_energy_2, ...
-
-    The contents of the input files should be organized as two columns separated by a space
-    and not contain any headers ex:
-        1.0 2.0
-        2.0 3.0
-        3.0 4.0
-        ...
-    the data reader is np.loadtxt()
-
-    left_col is the type of data in the left column ex: 'volume'
-    right_col is the type of data in the right column ex: 'energy'
-    """
-    df = pd.DataFrame(columns=["config", left_col, right_col])
-    for input_file in input_files:
-        config = os.path.splitext(os.path.basename(input_file))[0]
-        if "_" in config:
-            config = config.split("_")[-1]
-        data = np.loadtxt(input_file)
-        left_data = data[:, 0]
-        right_data = data[:, 1]
-        for i in range(len(left_data)):
-            df.loc[len(df)] = {
-                "config": config,
-                left_col: left_data[i],
-                right_col: right_data[i],
-            }
-    return df
-
-
-# TODO: review
-def select_data(df: pd.DataFrame, selection_dict: dict) -> pd.DataFrame:
-    """
-    Used to cherry pick data from a dataframe based on the config name and the volume rank
-    the selection_dict should be a dictionary with the following format:
-    selection_dict = {10: [0,1,2,3],
-                        11: [0,1,2,3],
-                        12: [0,1,2,3],
-                        ...}
-    where the keys are the config names and the values are the volumes 0=lowest
-    volume, 1=second lowest volume, etc.
-
-
-    """
-    # first rank the volumes from lowest to highest
-    df["volume_rank"] = df.groupby("config")["volume"].rank(method="dense").astype(int)
-    print(df)
-    selected_data = pd.DataFrame()  # Initialize selected_data variable
-    for config in selection_dict.keys():
-        for volume_rank in selection_dict[config]:
-            selected_data = pd.concat(
-                [
-                    selected_data,
-                    df[(df["config"] == config) & (df["volume_rank"] == volume_rank)],
-                ],
-                ignore_index=True,
-            )
-    return selected_data
-
-
-# TODO: review
+# TODO: Not sure if a list of dataframe will actually work. the function simply concats them all together
 def plot_mv(df: pd.DataFrame, show_fig: bool = True) -> go.Figure:
-    """
-    data may be a single pandas data frame or a list of pandas data frames
-    data may also be a list of input_file names as strings ex:
-        ['str_0', 'str_1', 'str_2', ...]
+    """Plot the magnetic moment vs volume
 
-    Not sure if a list of dataframe will actually work. the function simply
-    concats them all together # TODO: test this
```

Comment:
It doesn't no need to test. If I wrote this, I lied. The concatenation happens for the nested dataframe containing the magnetic data.

Reply:
Ok. Thanks.